src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -816,12 +816,12 @@ void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile() {
                                           &(msg_params[strings::app_hmi_type]));
   }
 #endif  // SDL_REMOTE_CONTROL
+  SendResponse(true, result_code, add_info.c_str(), &response_params);
+
   // Default HMI level should be set before any permissions validation, since it
   // relies on HMI level.
   application_manager_.OnApplicationRegistered(application);
 
-  SendResponse(true, result_code, add_info.c_str(), &response_params);
```

Comment:
@tkireva also note that **BC.ActivateApp should be called after OnAppRegistered**, please also be ensured in this behavior.